### PR TITLE
Add retaskable + retaskable-capture

### PR DESCRIPTION
**Pull request by jdkruzr**
CalDAV task client for reMarkable tablets, plus its xochitl note-capture/jump-back hooks. The app is source-built from the reTaskable `v1.0.2` GitHub tag for both aarch64 and armv7; capture hooks are OS-pinned to 3.27 and depend on the app.

### CODE REVIEW

File: packages/retaskable-capture/VELBUILD [hunk 1]
```diff
@@ -0,0 +1,68 @@
+maintainer="Jarett <jarett@reticulum.us>"
+pkgname=retaskable-capture
+pkgver=1.0.0
+pkgrel=0
+upstream_author="jdkruzr"
+category="ui"
+pkgdesc="reTaskable note capture: lasso handwriting into a to-do and jump back to the source page (patches xochitl via xovi+qmldiff)."
```
Comment (Eeems) [inline packages/retaskable-capture/VELBUILD]:
```suggestion
pkgdesc="reTaskable note capture: lasso handwriting into a to-do and jump back to the source page."
```
User's don't care that it patches xochitl.

File: packages/retaskable-capture/VELBUILD [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+url="https://github.com/jdkruzr/reTaskable"
+arch="noarch"
+license="GPL-3.0-only"
+# Depends on the app, the qmldiff host, and the two asivery extensions the hooks
+# use (capture auto-sync + jump-back IPC). NOTE the package is qt-command-executor,
+# NOT command-executor. The OS pin is load-bearing: these hooks are verified only
+# against xochitl 3.27's QML, and a qmldiff that fails to apply makes
+# qt-resource-rebuilder safe-fail ALL mods — so we refuse to install on an
+# unverified OS rather than risk that. !convert-to-text-remover because our capture
+# reuses xochitl's convert-to-text recognition path, which that mod strips.
```
Comment (Eeems) [inline packages/retaskable-capture/VELBUILD]:
```suggestion
```

File: packages/retaskable-capture/VELBUILD [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+depends="retaskable qt-resource-rebuilder qt-command-executor xovi-message-broker !convert-to-text-remover remarkable-os>=3.27 remarkable-os<3.28"
```
Comment (Eeems) [inline packages/retaskable-capture/VELBUILD]:
```suggestion
depends="
retaskable
qt-resource-rebuilder
qt-command-executor
xovi-message-broker
!convert-to-text-remover
remarkable-os>=3.27
remarkable-os<3.28
"
```

File: packages/retaskable-capture/VELBUILD
Comment (Eeems) [inline packages/retaskable-capture/VELBUILD:1]: Did you explore using a subpackage instead of two separate VELBUILD files?

File: packages/retaskable-capture/VELBUILD [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+options="!check !fhs"
+readmeurl="https://raw.githubusercontent.com/$upstream_author/reTaskable/v$pkgver/README.md"
+
+# The hooks are plain text shipped straight from the repo at the release tag, in
+# their committed HASHED form — which is exactly what the rest of the Vellum qmd
+# ecosystem ships (bettertoc, quicksettings-screenshot, etc. are all hashed). The
+# hashed tokens are deterministic identifier hashes that qt-resource-rebuilder
+# matches via the device's hashtab (regenerated by xovi/rebuild_hashtable), so
+# they're portable across same-version devices. No unhash step needed.
```
Comment (Eeems) [inline packages/retaskable-capture/VELBUILD]:
```suggestion
```

File: packages/retaskable-capture/VELBUILD [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+	echo ""
+}
+
+postosupgrade() {
+	echo ""
+	echo "  NOTE: reTaskable capture patches xochitl and is pinned to a"
+	echo "  specific OS range. After an OS upgrade it may need an update;"
+	echo "  if note-capture stops working, check for a newer version."
+	echo ""
+}
```
Comment (Eeems) [inline packages/retaskable-capture/VELBUILD]:
```suggestion
```
This kind of message would be way too much noise if every single xovi based modification provided it.

File: packages/retaskable-capture/VELBUILD [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+		"$pkgdir"/home/root/xovi/exthome/qt-resource-rebuilder/retaskableSend.qmd
+	install -Dm644 "$srcdir"/retaskableJump.qmd \
+		"$pkgdir"/home/root/xovi/exthome/qt-resource-rebuilder/retaskableJump.qmd
+
+	install -Dm644 "$srcdir"/LICENSE \
+		"$pkgdir"/home/root/.vellum/licenses/$pkgname/LICENSE
+	echo "$url" > "$pkgdir"/home/root/.vellum/licenses/$pkgname/SOURCES
+}
+
+# Generated for the v1.0.0 release (sha512sum of the .qmd + LICENSE at the tag).
```
Comment (Eeems) [inline packages/retaskable-capture/VELBUILD]:
```suggestion
```
This is automatically generated by `vbuild checksum`. No comment necessary.

File: packages/retaskable/VELBUILD [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+pkgrel=0
+upstream_author="jdkruzr"
+category="utilities"
+pkgdesc="CalDAV (VTODO) task client for the reMarkable Paper Pro — sync to-dos, due dates, offline-first."
+url="https://github.com/jdkruzr/reTaskable"
+arch="aarch64"
+license="GPL-3.0-only"
+# The app is a sandboxed AppLoad app (it does NOT patch xochitl), so it only needs
+# the AppLoad runtime. appload itself carries the qt-resource-rebuilder + OS pins;
+# we mirror the OS range for clarity.
```
Comment (Eeems) [inline packages/retaskable/VELBUILD]:
```suggestion
```

File: packages/retaskable/VELBUILD [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+depends="appload remarkable-os>=3.26 remarkable-os<3.28"
```
Comment (Eeems) [inline packages/retaskable/VELBUILD]:
```suggestion
depends="appload"
```
If the only reason you have the OS versions listed is because appload has them upstream, it's not useful to add them here. Only add if this package has a specific OS version requirement, appload's depends will handle the rest.

File: packages/retaskable/VELBUILD [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+options="!check !fhs !strip !tracedeps"
+readmeurl="https://raw.githubusercontent.com/$upstream_author/reTaskable/v$pkgver/README.md"
+
+# PREBUILT distribution: cut a GitHub release tagged v$pkgver with the output of
+# app/build-rmpp.sh zipped as retaskable-aarch64.zip (the tree: backend/entry,
+# resources.rcc, manifest.json, icon.png). See packaging/vellum/README.md.
```
Comment (Eeems) [inline packages/retaskable/VELBUILD]:
```suggestion
```

File: packages/retaskable/VELBUILD [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+source="
+https://github.com/$upstream_author/reTaskable/releases/download/v$pkgver/retaskable-aarch64.zip
```
Comment (Eeems) [inline packages/retaskable/VELBUILD]: Is building with vbuild missing something you need to get it to work? This would also help add armv7 builds for the rM1/2.

File: packages/retaskable/VELBUILD [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+https://raw.githubusercontent.com/$upstream_author/reTaskable/v$pkgver/LICENSE
+"
+
+unpack() {
+	unzip -o "$srcdir"/retaskable-aarch64.zip -d "$srcdir"
```
Comment (Eeems) [inline packages/retaskable/VELBUILD]: What was the reason to override the default unpack?

File: packages/retaskable/VELBUILD [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+		"$pkgdir"/home/root/xovi/exthome/appload/retaskable/manifest.json
+	install -Dm644 "$srcdir"/icon.png \
+		"$pkgdir"/home/root/xovi/exthome/appload/retaskable/icon.png
+
+	install -Dm644 "$srcdir"/LICENSE \
+		"$pkgdir"/home/root/.vellum/licenses/$pkgname/LICENSE
+	echo "$url" > "$pkgdir"/home/root/.vellum/licenses/$pkgname/SOURCES
+}
+
+# Generated for the v1.0.0 release (sha512sum of the published artifacts).
```
Comment (Eeems) [inline packages/retaskable/VELBUILD]:
```suggestion
```

**Review by Eeems — CHANGES_REQUESTED** [review]

File: packages/retaskable-capture/VELBUILD
Comment (jdkruzr) [reply packages/retaskable-capture/VELBUILD:1]: nope, this is my first rM app so I didn't consider anything but the simplest cases. I think you could make an argument that a user could conceivably want to run rT without the selection bar plugin, though, so maybe that changes the calculus?

File: packages/retaskable/VELBUILD (hunk 8 above)
Comment (jdkruzr) [reply packages/retaskable/VELBUILD]: good to know, ty.

File: packages/retaskable-capture/VELBUILD
Comment (Eeems) [reply packages/retaskable-capture/VELBUILD:1]: You should be able to use a single VELBUILD with a subpackage defined and have the second package inside the subpackage. One of them can then depend on the other. That way things stay together, and since they are pulling from the same source repo it can simplify how you pull things.

**Comment by jdkruzr** [comment]
Updated in 553a39d:

- Folded capture into a `retaskable-capture` subpackage inside the main `retaskable` VELBUILD.
- Switched the app package from the prebuilt release zip to a source build from upstream `v1.0.1`.
- Added armv7 to the app build path alongside aarch64. I can't personally validate rm1/rm2 hardware, so testing from someone with those devices would still be welcome.
- Kept capture available without rm1/rm2 exclusions and pinned only to `remarkable-os>=3.27`; the hooks were developed/verified against xochitl 3.27 QML, but I'm intentionally not adding a `<3.28` upper bound unless breakage is reported.
- Applied the cleanup suggestions: shorter descriptions, multiline deps, no noisy lifecycle messages, no custom unpack, and no checksum comments.

Validation now includes:

- `vbuild -C packages/retaskable checksum`
- `vbuild -C packages/retaskable gen`
- `./scripts/build-package.sh retaskable aarch64`
- `./scripts/build-package.sh retaskable armv7`

Those produced both `retaskable` and `retaskable-capture` APKs for aarch64 and armv7. The upstream Rust tests also pass (138 tests), and the upstream aarch64 SDK build was validated from the reTaskable repo root.

**Comment by jdkruzr** [comment]
Follow-up correction: I remembered the local reTaskable SDK path and validated the upstream aarch64 build too.

Validation also includes:

- From the reTaskable repo root: `source sdk/environment-setup-cortexa55-remarkable-linux && ./app/build-rmpp.sh`
- Produced `app/output-rmpp/backend/entry` as an aarch64 ELF plus `resources.rcc`, `manifest.json`, and `icon.png`.
- Updated upstream `v1.0.1` so `build-rmpp.sh` works with both the local SDK compiler names and the Vellum/Toltec compiler names.
- Updated this PR source archive checksum to match the final `v1.0.1` tag.
- Validated the Vellum package builds through vbuild/Podman for both aarch64 and armv7.

I still have not personally hardware-tested rm1/rm2.

**Comment by jdkruzr** [comment]
[@Eeems] anything else you need from me? I actually do need to push up a small bugfix change anyways.

**Comment by Eeems** [comment]
Push away, I just haven't had time to test and merge yet.

**Comment by jdkruzr** [comment]
Pushed `7d63066`, bumping `retaskable` to upstream `v1.0.2`.

What changed upstream in `v1.0.2`:

- Fixes a sync edge case where a full reconciliation could delete a still-pending local create row, leaving the create op to report `cache row vanished mid-flush` even after the server accepted the task.
- Adds a Settings sync-error panel that lists errored/retrying pending operations with op id, type, summary, age, attempt count, and full last error for issue reports.

Validation rerun:

- `cargo test --bin retaskable-backend` in reTaskable: 142 passed.
- `vbuild -C packages/retaskable checksum`
- `vbuild -C packages/retaskable gen && vbuild -C packages/retaskable validate`
- `CARCH=aarch64 vbuild -C packages/retaskable clean && CARCH=aarch64 vbuild -C packages/retaskable all`
- `CARCH=armv7 vbuild -C packages/retaskable clean && CARCH=armv7 vbuild -C packages/retaskable all`

Those produced `retaskable` and `retaskable-capture` APKs for both aarch64 and armv7 from the pushed `v1.0.2` source archive. Same caveat as before: I still have not personally hardware-tested rm1/rm2.

**Review by Eeems — CHANGES_REQUESTED** [review]
- icon is just the word "example"
- The title and "not yet synced" text is rather small and hard to read compared to the button text
- The "no open tasks" text is quite a faint grey, and hard to read.
- There is no way to show the entered password to make sure it was entered correctly, which is rather painful
- I can't appear to connect to my nextcloud instance, it's complaining about "step 1: PROPFIND"  and "No address associated with hostname". So I can't validate if it's actually working.
- I can try to create a task and click the create button but nothing is added, which is likely due to the lack of synchronization setup

**Comment by jdkruzr** [comment]
Pushed `c2d2a21`, bumping `retaskable` to upstream `v1.0.3`.

This addresses the testing feedback:

- Replaced the placeholder `EXAMPLE` app icon with a simple check/task mark matching the capture affordance.
- Increased the app title/status/empty-state readability and darkened the empty-state text.
- Added a Show/Hide control for the Settings app-password field.
- Made Create visibly unavailable until Settings + first Sync have established the target calendar; tapping with a draft summary now gives a setup/sync prompt instead of looking like a no-op.
- Normalized common Server URL inputs (`nextcloud.example.com`, bare `/remote.php/dav`) and made DNS / malformed URL / auth discovery errors clearer while preserving the underlying detail.

Validation rerun:

- Upstream `cargo test --bin retaskable-backend`: 146 passed.
- Upstream aarch64 SDK build: `source sdk environment && ./app/build-rmpp.sh` produced `backend/entry`, `resources.rcc`, `manifest.json`, and the new `icon.png`.
- `vbuild -C packages/retaskable checksum`
- `vbuild -C packages/retaskable gen && vbuild -C packages/retaskable validate`
- `CARCH=aarch64 vbuild -C packages/retaskable clean && CARCH=aarch64 vbuild -C packages/retaskable all`
- `CARCH=armv7 vbuild -C packages/retaskable clean && CARCH=armv7 vbuild -C packages/retaskable all`

Both Vellum builds produced `retaskable` and `retaskable-capture` APKs for `aarch64` and `armv7`. Same caveat as before: I still have not personally hardware-tested rm1/rm2.

**Comment by jdkruzr** [comment]
Pushed `0dbbc55`, bumping `retaskable` to upstream `v1.0.4`.

This is a small icon-only follow-up after device testing: the app icon is now a strictly two-color 100x100 PNG with no antialiasing/alpha fringe, which avoids rendering artifacts in the AppLoad launcher.

Validation rerun:

- Upstream icon was installed on-device and verified by checksum against `/home/root/xovi/exthome/appload/retaskable/icon.png`.
- `vbuild -C packages/retaskable checksum`
- `vbuild -C packages/retaskable gen && vbuild -C packages/retaskable validate`
- `CARCH=aarch64 vbuild -C packages/retaskable clean && CARCH=aarch64 vbuild -C packages/retaskable all`
- `CARCH=armv7 vbuild -C packages/retaskable clean && CARCH=armv7 vbuild -C packages/retaskable all`

Both Vellum builds produced `retaskable` and `retaskable-capture` APKs for `aarch64` and `armv7` from the pushed `v1.0.4` source archive.

Also noting for testing-repo users: the earlier `/publish-testing` run happened before the `v1.0.3` and `v1.0.4` bumps, so it will need to be run again if the testing package should include the latest review fixes and cleaned-up icon.

**Review by Eeems — CHANGES_REQUESTED** [review]
- It's now complaining about "No 'Authorization: Basic' header found". When I try to go to the url it mentions, if I use the same credentials requested for basic auth, I do not receive a 401.
- I still have concerns with how difficult it can be to read much of the text

**Comment by jdkruzr** [comment]
Pushed `478eedb`, bumping `retaskable` to upstream `v1.0.5`.

This addresses the latest testing feedback:

- Hardened CalDAV discovery redirects: WebDAV `PROPFIND`/`REPORT` requests now use explicit redirect handling and reapply Basic auth on the redirected request. This should cover servers whose `/.well-known/caldav` endpoint redirects before returning DAV properties, which matches the `No 'Authorization: Basic' header found` symptom.
- Added a regression test for redirected discovery to assert the redirected `PROPFIND` still carries `Authorization: Basic ...`.
- Made another e-ink readability pass across the app: larger/darker status text, task metadata, Settings labels/fields/status/errors, and due-date controls; also removed monospace from cramped user-facing status/error areas.

Validation rerun:

- Upstream `cargo test --bin retaskable-backend`: 147 passed.
- Upstream aarch64 SDK build: `source sdk environment && ./app/build-rmpp.sh` produced `backend/entry`, `resources.rcc`, `manifest.json`, and `icon.png`.
- Upstream live discovery check through the app backend protocol against a configured Nextcloud account: succeeded, 3 calendars discovered.
- `vbuild -C packages/retaskable checksum`
- `vbuild -C packages/retaskable gen && vbuild -C packages/retaskable validate`
- `CARCH=aarch64 vbuild -C packages/retaskable clean && CARCH=aarch64 vbuild -C packages/retaskable all`
- `CARCH=armv7 vbuild -C packages/retaskable clean && CARCH=armv7 vbuild -C packages/retaskable all`

Both Vellum builds produced `retaskable` and `retaskable-capture` APKs for `aarch64` and `armv7` from the pushed `v1.0.5` source archive.

**Review by Eeems — APPROVED** [review]
